### ISSUE TITLE
Segmentation fault on linux' logo.gif

### ISSUE THREAD

**Issue by walt-ko** [IlyaGrebnov/bsc-m03#4]
file linux/Documentation/logo.gif from every linux kernel source starting from linux-2.0.tar
bsc-m03 is compiled for 64-bit arch (does not reproduced if compiled for 32-bit arch)

```
walt@myhost:/dev/shm$ /usr/local/src/compression/bsc-m03/bsc-m03-0.2.1/bsc-m03 e logo.gif logo.gif.m03-021
bsc-m03 is experimental block sorting compressor. Version 0.2.1 (08 January 2022).
Copyright (c) 2021-2022 Ilya Grebnov <Ilya.Grebnov@gmail.com>. ABSOLUTELY NO WARRANTY.
This program is based on (at least) the work of Michael Maniscalco (see AUTHORS).

logo.gif compressed from 16335 into 16300 in 0.046 seconds (7.983 bps).
walt@myhost:/dev/shm$ /usr/local/src/compression/bsc-m03/bsc-m03-0.3.0/bsc-m03 e logo.gif logo.gif.m03-030
bsc-m03 is experimental block sorting compressor. Version 0.3.0 (10 November 2022).
Copyright (c) 2021-2022 Ilya Grebnov <Ilya.Grebnov@gmail.com>. ABSOLUTELY NO WARRANTY.
This program is based on (at least) the work of Michael Maniscalco (see AUTHORS).

logo.gif compressed from 16335 into 16281 in 0.046 seconds (7.974 bps).
walt@myhost:/dev/shm$ /usr/local/src/compression/bsc-m03/bsc-m03-0.4.0/bsc-m03 e logo.gif logo.gif.m03-040
bsc-m03 is experimental block sorting compressor. Version 0.4.0 (20 November 2022).
Copyright (c) 2021-2022 Ilya Grebnov <Ilya.Grebnov@gmail.com>. ABSOLUTELY NO WARRANTY.
This program is based on (at least) the work of Michael Maniscalco (see AUTHORS).

Compressing logo.gif(00%)Segmentation fault
walt@myhost:/dev/shm$ /usr/local/src/compression/bsc-m03/bsc-m03-0.5.0/bsc-m03 e logo.gif logo.gif.m03-050
bsc-m03 is experimental block sorting compressor. Version 0.5.0 (27 November 2022).
Copyright (c) 2021-2022 Ilya Grebnov <Ilya.Grebnov@gmail.com>. ABSOLUTELY NO WARRANTY.
This program is based on (at least) the work of Michael Maniscalco (see AUTHORS).

Compressing logo.gif(00%)Segmentation fault
walt@myhost:/dev/shm$ ls -l logo.gif*
-rw-r--r-- 1 walt walt 16335 May  7 23:24 logo.gif
-rw-rw-r-- 1 walt walt 16300 May  8 00:40 logo.gif.m03-021
-rw-rw-r-- 1 walt walt 16281 May  8 00:41 logo.gif.m03-030
-rw-rw-r-- 1 walt walt     0 May  8 00:41 logo.gif.m03-040
-rw-rw-r-- 1 walt walt     0 May  8 00:41 logo.gif.m03-050
walt@myhost:/dev/shm$ /usr/local/src/compression/bsc-m03/bsc-m03-0.2.1/bsc-m03 d logo.gif.m03-021 logo.gif_021
bsc-m03 is experimental block sorting compressor. Version 0.2.1 (08 January 2022).
Copyright (c) 2021-2022 Ilya Grebnov <Ilya.Grebnov@gmail.com>. ABSOLUTELY NO WARRANTY.
This program is based on (at least) the work of Michael Maniscalco (see AUTHORS).

logo.gif.m03-021 decompressed from 16300 into 16335 in 0.047 seconds.
walt@myhost:/dev/shm$ /usr/local/src/compression/bsc-m03/bsc-m03-0.3.0/bsc-m03 d logo.gif.m03-030 logo.gif_030
bsc-m03 is experimental block sorting compressor. Version 0.3.0 (10 November 2022).
Copyright (c) 2021-2022 Ilya Grebnov <Ilya.Grebnov@gmail.com>. ABSOLUTELY NO WARRANTY.
This program is based on (at least) the work of Michael Maniscalco (see AUTHORS).

logo.gif.m03-030 decompressed from 16281 into 16335 in 0.048 seconds.
walt@myhost:/dev/shm$ sha256sum -b logo.gif logo.gif_*
4cdf8d34e001fc7f15b61823eee5617f5389e153d7d317471d0f9d982c0a2745 *logo.gif
4cdf8d34e001fc7f15b61823eee5617f5389e153d7d317471d0f9d982c0a2745 *logo.gif_021
4cdf8d34e001fc7f15b61823eee5617f5389e153d7d317471d0f9d982c0a2745 *logo.gif_030
walt@myhost:/dev/shm$ 

```

**Comment by IlyaGrebnov** [comment]
@walt-ko could you please share details how you compiled bsc-m03? So far I was not able to reproduce this.

**Comment by IlyaGrebnov** [comment]
@walt-ko I have repro. I am going to look into this.

**Comment by IlyaGrebnov** [comment]
Fixed in version 0.5.5. Thank you for reporting this issue.

**Comment by walt-ko** [comment]
Now it works. Thnx
